drivers/pinmux/stm32/pinmux_stm32.c

```diff
@@ -127,15 +127,35 @@ int stm32_dt_pinctrl_configure(const struct soc_gpio_pinctrl *pinctrl,
 			       STM32_DT_PINMUX_LINE(mux));
 		port_device = gpio_ports[STM32_PORT(pin)];
 
+#ifdef CONFIG_PM_DEVICE_IDLE
+		/* FIXME: Use of device_pm_foo clock mgmt function */
+		/* breaks correct gpio driver behavior */
+		/* Use direct clock gatting function as a workaround */
+		/* BEWARE that this has serious limitations in case */
+		/* of multithreaded application */
+		/* ret = device_pm_get_sync(port_device); */
```

Comment:
what happens here ? it seems the key point of having a device pm interface.

Reply:
As mentioned in the commit message, using `device_pm_get_sync` here produces strange gpio driver behavior (test/drivers/gpio/gpio_basic_api failing).
So, indeed this should be fixed before merging. I'll try to investigate more on this today.